catris/squares.py

```diff
@@ -143,22 +205,72 @@ def get_text(self, landed: bool) -> bytes:
  \/
 
 """
+DRILL_PICTURES_LEFT = rb"""
+
+ .--------
+'._\__\__\
+
+ .--------
+'.__\__\__
+
+ .--------
+'.\__\__\_
+
+"""
+DRILL_PICTURES_RIGHT = rb"""
+
+--------.
+_/__/__/.'
+
+--------.
+/__/__/_.'
+
+--------.
+__/__/__.'
+
+"""
 
 
 class DrillSquare(Square):
     def __init__(self) -> None:
         super().__init__(_RotateMode.NO_ROTATING)
-        self.picture_x = 0
-        self.picture_y = 0
         self.picture_counter = 0
 
-    def get_text(self, landed: bool) -> bytes:
-        picture_list = DRILL_PICTURES.strip().split(b"\n\n")
+    def get_text(self, player: Player | None, landed: bool) -> bytes:
+        dir_x = self.moving_dir_x
+        dir_y = self.moving_dir_y
+        if self._in_world_coordinates:
+            assert player is not None
+            dir_x, dir_y = player.world_to_player(dir_x, dir_y)
+
+        def rotate(x: int, y: int) -> tuple[int, int]:
+            # Trial and error was used to figure out some of this
+            return (dir_y * x + dir_x * y, -dir_x * x + dir_y * y)
+
+        x, y = rotate(self.original_x, self.original_y)
+        corners = {
+            rotate(-1, 0),
+            rotate(-1, 1 - DRILL_HEIGHT),
+            rotate(0, 0),
+            rotate(0, 1 - DRILL_HEIGHT),
+        }
+        relative_x = x - min(x for x, y in corners)
+        relative_y = y - min(y for x, y in corners)
+
+        picture_list = (
+            {
+                (0, -1): DRILL_PICTURES_UP,
+                (0, 1): DRILL_PICTURES_DOWN,
+                (-1, 0): DRILL_PICTURES_LEFT,
+                (1, 0): DRILL_PICTURES_RIGHT,
+            }[dir_x, dir_y]
+            .strip(b"\n")
+            .split(b"\n\n")
+        )
```

Comment:
Would it be simpler to just put the results of all this to variables, instead of parsing somewhat complex strings?